pkgs/by-name/ve/veridian/package.nix

```diff
@@ -0,0 +1,155 @@
+{
+  stdenv,
+  rustPlatform,
+  fetchFromGitHub,
+  cmake,
+  python3,
+  ninja,
+  pkg-config,
+  verilator,
+  verible,
+  fmt,
+  boost182,
+  sv-lang,
+  catch2_3,
+  openssl,
+  lib,
+  writeText,
+}:
+
+let
+  fmtver = "11.1.3";
+  fmtsrc = fetchFromGitHub {
+    owner = "fmtlib";
+    repo = "fmt";
+    tag = fmtver;
+    hash = "sha256-6r9D/csVSgS+T/H0J8cSR+YszxnH/h2V2odi2s6VYN8=";
+  };
+
+  fmtlib = fmt.overrideAttrs (oldAttrs: {
+    version = fmtver;
+    src = fmtsrc;
+  });
+
+  boostlib = boost182;
+
+  slang-git =
+    let
+      version = "8.0";
+    in
+    sv-lang.overrideAttrs (old: {
+      version = version;
+      src = fetchFromGitHub {
+        owner = "MikePopoloski";
+        repo = "slang";
+        tag = "v${version}";
+        hash = "sha256-UZwMxnzprOjN5DlV1zLd0I6rgrBMBhcZ+dSx8t2sJF8=";
+      };
+      cmakeFlags = [
+        "-DCMAKE_INSTALL_INCLUDEDIR=include"
+        "-DCMAKE_INSTALL_LIBDIR=lib"
+        "-DSLANG_INCLUDE_TESTS=${if stdenv.hostPlatform.isDarwin then "OFF" else "ON"}"
+        "-DSLANG_USE_MIMALLOC=OFF"
+        "-DFETCHCONTENT_SOURCE_DIR_FMT=${fmtsrc}"
+      ];
+      nativeBuildInputs = [
+        cmake
+        python3
+        ninja
+      ];
+      buildInputs = [
+        (stdenv.mkDerivation rec {
+          pname = "unordered_dense";
+          version = "2.0.1";
+          src = fetchFromGitHub {
+            owner = "martinus";
+            repo = "unordered_dense";
+            tag = "v${version}";
+            hash = "sha256-9zlWYAY4lOQsL9+MYukqavBi5k96FvglRgznLIwwRyw=";
+          };
+          nativeBuildInputs = [ cmake ];
+          installPhase = "runHook preInstall; mkdir -p $out/include; cp -r . $out/include";
+        })
+        boostlib
+        fmtlib
+        catch2_3
+      ];
+    });
+in
+rustPlatform.buildRustPackage {
+  pname = "veridian";
+  version = "0-unstable-2024-12-25";
+  src = fetchFromGitHub {
+    owner = "vivekmalneedi";
+    repo = "veridian";
+    rev = "d094c9d2fa9745b2c4430eef052478c64d5dd3b6";
+    hash = "sha256-3KjUunXTqdesvgDSeQMoXL0LRGsGQXZJGDt+xLWGovM=";
+  };
+
+  cargoHash = "sha256-qJQD9HjSrrHdppbLNgLnXCycgzbmPePydZve3A8zGtU=";
+  useFetchCargoVendor = true;
+
+  buildFeatures = [ "slang" ];
+
+  nativeBuildInputs = [
+    rustPlatform.bindgenHook
+    pkg-config
+    cmake
+    verilator
+    verible
+  ];
+
+  buildInputs = [
+    fmtlib
+    verible
+    slang-git
+  ];
+
+  patches = [
```

Comment:
Patches are usually placed next to the nix file into `.patch`  files.
If you also want to substitute some values into the patch file, consider using the `replaceVars` helper.
(Look at other packages for examples on how to use it.)